### PR TITLE
add support where methods with opt params

### PR DESCRIPTION
**Pull request by huuhait**
# What does this PR do?

**QueryBuilder**
Make a query like rails `User.where(first_name: "John")`

### CODE REVIEW

File: src/jennifer/query_builder/query.cr [hunk 1]
```diff
@@ -301,6 +301,21 @@ module Jennifer
         self
       end
 
+      # Mutates query applying all modification returned from the block.
```
Comment (imdrasil) [inline src/jennifer/query_builder/query.cr:304]: It seems docs doesn't describe the real behavior

File: src/jennifer/query_builder/query.cr [hunk 2]
```diff
@@ -301,6 +301,21 @@ module Jennifer
...
+      #
+      # Yields the expression builder and block is also executed with expression builder context.
+      #
+      # ```
+      # User.where(email: "example@test.com")
+      # ```
+      def where(**opts)
```
Comment (imdrasil) [inline src/jennifer/query_builder/query.cr:311]: we should support hashes as well

File: src/jennifer/query_builder/query.cr [hunk 3]
```diff
@@ -301,6 +301,21 @@ module Jennifer
...
+        data = self
+        opts.each do |k, v|
+          data = data.where { c(k.to_s) == v }
```
Comment (imdrasil) [inline src/jennifer/query_builder/query.cr:314]: we can avoid N invocation of `#where` here iterating over key-values inside of the block; also `data` variable seems unnecessary here as it is just an alias for `self`

File: src/jennifer/model/resource.cr [hunk 4]
```diff
@@ -221,6 +221,19 @@ module Jennifer
         {% end %}
       end
 
+      # Is a shortcut for `.all.where` call.
+      #
+      # ```
+      # User.where(name: "John")
+      # ```
+      def self.where(**opts)
```
Comment (imdrasil) [inline src/jennifer/model/resource.cr:229]: let's delegate this call to `#where`

```crystal
def self.where(*args, **opts)
  all.where(*args, **opts)
end
```

File: src/jennifer/query_builder/query.cr (hunk 2 above)
Comment (imdrasil) [reply src/jennifer/query_builder/query.cr:311]: also would you mind to add tests

Comment (huuhait) [reply src/jennifer/query_builder/query.cr:309]: @imdrasil i'm not good at test :( sorry

**Pull request closed without merging**